prow/apis/prowjobs/v1/types.go

```diff
@@ -646,6 +647,30 @@ func (d *DecorationConfig) Validate() error {
 	return nil
 }
 
+const (
+	logMountName            = "logs"
+	codeMountName           = "code"
+	toolsMountName          = "tools"
+	gcsCredentialsMountName = "gcs-credentials"
+	s3CredentialsMountName  = "s3-credentials"
+)
+
+// Volumes returns a set of secret names that will eventually be created in the pod as volumes.
+func (d *DecorationConfig) Volumes() sets.String {
+	ret := sets.NewString(logMountName, codeMountName, toolsMountName, gcsCredentialsMountName, s3CredentialsMountName)
+	if d == nil {
+		return ret
```

Comment:
This is wrong, if the decoration config is nil it has to return nil.

Reply:
updated